conda/core/index.py

```diff
@@ -331,7 +332,53 @@ def maybe_decompress(filename, resp_content):
                              getattr(e.response, 'elapsed', None))
 
 
-def read_local_repodata(cache_path):
+def write_pickled_repodata(cache_path, repodata):
+    # Don't bother to pickle empty channels
+    if not repodata.get('packages'):
+        return
+    try:
+        import pickle
+        with open(cache_path + '.q', 'wb') as f:
+            pickle.dump(repodata, f)
+    except:
+        pass
+
+
+def read_picked_repodata(cache_path, channel_url, schannel, priority):
+    pickle_path = cache_path + '.q'
+    # Don't trust pickled data if there is no accompanying json data
+    if not isfile(pickle_path) or not isfile(cache_path):
+        return None
+    try:
+        import pickle
+        with open(cache_path + '.q', 'rb') as f:
+            repodata = pickle.load(f)
+    except:
+        return None
+    if repodata.get('_pversion') != REPODATA_PICKLE_VERSION:
+        return None
+    if repodata.get('_url') != channel_url:
```

Comment:
I don't know if this is necessary or not, but just to be safe, if for any reason the URL is changed I don't trust this data.